rc/core/kakrc.kak

```diff
@@ -15,25 +15,38 @@ add-highlighter -group / regions -default code kakrc \
     comment (^|\h)\K\# $ '' \
     double_string %{(^|\h)"} %{(?<!\\)(\\\\)*"} '' \
     single_string %{(^|\h)'} %{(?<!\\)(\\\\)*'} '' \
-    shell '%sh\{' '\}' '\{'
```

Comment:
Any punctuation sign can be used for scopes, including `!`, `~`, `#` etc. I've seen those being used in some scripts already, so do we aim at providing exhaustive support for all types of scopes here? If yes (I don't know if it's worth it or not), I think these lines could be generated for a shell `for` loop.